qiskit/backends/_sympy_unitarysimulator.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""Contains a (slow) Python simulator that returns the unitary of the circuit.
+
+It produces the unitary of a quantum circuit in the symbolic form.
+In particular, it simulates the quantum computation with the sympy APIs,
+which preserve the symbolic form of numbers, e.g., sqrt(2), e^{i*pi/2}.
+
+How to use this simulator:
+see examples/python/use_sympy_simulators.py
+
+Example output:
+[[sqrt(2)/2 sqrt(2)/2 0 0]
+ [0 0 sqrt(2)/2 -sqrt(2)/2]
+ [0 0 sqrt(2)/2 sqrt(2)/2]
+ [sqrt(2)/2 -sqrt(2)/2 0 0]]
+
+Warning: it is slow.
+"""
+import uuid
+import logging
+import numpy as np
+from sympy.matrices import eye, zeros
+from sympy.physics.quantum import TensorProduct
+from sympy import Matrix, pi, E, I, cos, sin, N
+from qiskit._result import Result
+from qiskit.backends._basebackend import BaseBackend
+
+logger = logging.getLogger(__name__)
+
+
+def index1(barg, iarg, karg):
+    """Magic index1 function.
```

Comment:
Let's avoid calling things "magic".

Reply:
sounds good. will get rid of it.
I copied index1() from _simulatortools.py. 
Some one might want to fix the comment there as well.

Reply:
Oh. I will change it there. Why not importing `index2` from this file?

Reply:
thanks, 
index1 and index2 were in _unitsimulator.py and later refactored into _simulatortools.py.
I copied the code before the refactoring happens.

Reply:
fixed. import index1/index2 from _simulatortools.py directly